src/SFA.DAS.FAT.Web/Views/CourseProviders/CourseProviders.cshtml

```diff
@@ -12,18 +17,196 @@
-                
         </div>
+
+        <div class="govuk-grid-column-two-thirds  govuk-!-padding-0">
+        <div class="govuk-grid-row das-bottom-margin">
+            <div class="govuk-grid-column-full govuk-!-padding-right-0">
+                <div class="govuk-grid-column-one-third govuk-!-padding-10">
+                    <p class="govuk-body govuk-!-font-weight-bold">@Model.TotalMessage</p>
+                </div>
+                @if (Model.Providers.Count > 0)
+                {
+                    <div class="govuk-grid-column-two-thirds govuk-!-padding-0">
+                        <p class="govuk-body govuk-!-text-align-right">
+                              
+                            <label class="govuk-label" for="sort">
+                                Sort results &nbsp;&nbsp;
+
+                                <select class="govuk-select" id="course-providers-orderby" name="OrderBy">
+                                    
+                                    @foreach (var orderBy in Model.ProviderOrderDropdown)
+                                    {
+                                        <option value="@orderBy.ProviderOrderBy" selected=@(orderBy.Selected ? "selected" : null)>@orderBy.Description</option>
+                                    }
+                                </select>
+                            </label>
+                        </p>
+                    </div>
+                }
+            </div>
+        </div>
+        @if (Model.Providers.Count == 0)
+        {
+            <div class="das-left-margin">
+                <p class="govuk-body">
+                    There are no training providers to match your search.
+                </p>
+                <p class="govuk-body">
+                    <ul class="govuk-list govuk-list--bullet">
+                        <li>clear the filters you've selected</li>
+                        <li>increase the distance that the apprentice can travel</li>
+                    </ul>
+                </p>
+                <p class="govuk-body">
+                    Change your search using the filters on this page or <a asp-route="@RouteNames.SearchCourses" class="govuk-link govuk-link--no-visited-state">start a new search</a>.
+                </p>
+                <hr/>
+            </div>
+        }
+        else
+        {
+            <div class="das-search-results course-provider-results-main">
+                <ol class="das-search-results__list">
+
+                    @foreach (var provider in Model.Providers)
+                    {
+                        var providerRouteData = new Dictionary<string, string>
+                        {
+                            { "location", Model.Location },
+                            { "id", Model.Id.ToString() },
+                            { "providerId", provider.Ukprn.ToString() },
+                            { "distance", Model.Distance }
+                        };
```

Comment:
Can this be populated in the view model so we can have tests around it?

Reply:
yes

Reply:
yes, I've put this in its own partial with its own viewmodel, that incorporates the ProviderRouteData